tests/bdd/volgroup_vgpattern_bdd.feature

```diff
@@ -0,0 +1,179 @@
+
+        
+      Scenario: test sc with volgroup present on 2 nodes with less space and allowed topology
+          Given a sc is created with volgroup parameter as "lvmvg"
+          And node1 and node2 is in allowed topology
+          When a pvc is created with "1.3G" size
+          Then pvc should be Pending
+          And lvmvolume cr should not be created
+          When node2 vg is extended to <1.5G>
+          Then pvc should be Bound
+          And lvmvolume cr should be created
+          And lvmvolume cr should have expected owner and vg value
+
+      Examples:
+        |  node        | vg          |  size    |
+        | node1        | lvmvg       |  1G      |
+        | node2        | lvmvg       |  1G      |
+        | node3        | lvmvgdiff   |  1G      |
+
+  Background:
+
+      Given a 3 lvmnode cluster is configured
+      And each lvm node <node> has <vg> and <vg1> with <size> and <size1> respectively 
+
+      Scenario: test sc with volgroup present on 2 nodeswith varying free space and allowed topology
+          Given a sc is created with volgroup parameter as “lvmvg“
+          And using WaitForFirstConsumer as binding mode
+          And pvc is created with size “1.5G“
+          Then lvmvolume cr should not be created
+          When an app is created referencing the pvc
+          Then pvc should be created
+          And App should be running
+          And lvmvolume cr should be Ready
+          And lvmvolume cr should have expected owner and vg value
+
+       Examples:
+            |  node      | vg         |  size      |  vg1       |   size1     |
+            | node1      | lvmvg      |  1G        | lvmvg1     |     2G      |
+            | node2      | lvmvg      |  1.6G      | lvmvg1     |     2G      |
+            | node3      | lvmvg      |  1G        | lvmvg1     |     2G      |
+      
+  Background:
+
+      Given a 3 lvmnode cluster is configured
+      And each node <node> has <vg> and <vg2> created with same size
+
+      Scenario: test sc with vgpattern matching vg present in all node
+          Given sc1 is created with vgpattern parameter as "lvmvg1.*"
+          And sc2 is created with vgpattern parameter as "lvmvg2.*"
+          And sc3 is created with vgpattern parameter as "lvmvg3*."
+          When pvc1 is created referencing sc1
+          And pvc2 is created referencing sc2
+          And pvc3 is created referencing sc3
+          Then pvc1 and pvc2 should be in Bound state
+          And pvc1 and pvc2 lvmvolume cr should be in expected vg and node
+          And pvc3 should be in Pending state
+
+  
+      Examples:
+        |  node        | vg           | vg2           |
+        | node1        | lvmvg11       | lvmvg21      |
+        | node2        | lvmvg12       | lvmvg22      |
+        | node3        | lvmvg13       | lvmvg23      |
+
+  Background:
+    Given a 3 node lvm localpv cluster is configured
+    And each node has vg named "lvmvg"
```

Comment:
we'll keep this vg only on the node having custom label, and not on each node.

Reply:
Should that matter? Even if its on other nodes also, We will validate eventually lvmvolume schedules on expected node.